virttest/qemu_vm.py

```diff
@@ -4504,3 +4504,36 @@ def balloon(self, size):
     def system_reset(self):
         """ Send system_reset to monitor"""
         return self.monitor.system_reset()
+
+    def hotplug(self, device):
```

Comment:
Well this could be misleading as "vm.hotplug" could mean just about anything. It could mean:

* hotplugging a cpu in system `echo 1 > /sys/devices/system/cpu/`
* hotplugging actual cpu in qemu
* hotplugging a usb device
* hotplugging extra qemu device (eg. chardev, yes as far as I know it's not possible, but it might be in the future)
* hotplugging PCI device (which is what this does)

Another problem is it's only limited to 'pci.0'. We might eventually want to hotplug to different buses as well. I'd suggest creating new `QPciHotplugBus` that would allow extra root ports. On supported platforms we'd add those instead of `QPciBus`. We might add a new method `hotplug` to the bus that would only prepare the device for hotplugging (`get_free_root_port` and modify the devices's `bus` and `busid`s) which would be followed by `simple_hotplug` where it'd be actually inserted.

Alternative solution could be to simply do `insert` followed by `hotplug`, I remember it was a tough call which should go first.